File: services/user-service/src/main/java/com/bookland/user/entity/Role.java
package com.bookland.user.entity;

import lombok.*;
import jakarta.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "role")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToMany(mappedBy = "roles")
    @Builder.Default
    private Set<User> users = new HashSet<>();

    @ManyToMany
    @JoinTable(
            name = "role_permission",
            joinColumns = @JoinColumn(name = "roleId"),
            inverseJoinColumns = @JoinColumn(name = "permissionId")
    )
    @Builder.Default
    private Set<Permission> permissions = new HashSet<>();
}

File: BookLand_FE/src/types/Role.ts
import type { User } from './User.ts';
import type { Permission } from './Permission.ts';

export interface Role {
    id: number;
    name: string;
    description?: string;
    users?: User[];
    permissions?: Permission[];
}

export interface RoleRequest {
    name: string;
    description?: string;
}

export interface RoleQueryParams {
    keyword?: string;
    page?: number;
    size?: number;
    sortBy?: string;
    sortDirection?: string;
}
